databases/functions/userFollowUser/user-unfollow.js

```diff
@@ -45,5 +39,48 @@ module.exports = async (
       },
       {transaction: t}
     );
+  }
+};
+
+/**
+ *
+ * @param {Model} model
+ * @param {RegisterBodyData.Users} users
+ */
+module.exports = async (
+  userFollowUserModel,
+  userFollowUserHistoryModel,
+  userId,
+  followedUser = null,
+  followSource = '',
+  _t = null
+) => {
+  if (!followedUser) return;
+  await sequelize.transaction(async (t) => {
+    //unfoll sign user
+    await unfollUser(
+      userFollowUserModel,
+      userFollowUserHistoryModel,
+      userId,
+      followedUser,
+      followSource,
+      t
+    );
+    //end unfoll sign user
+
+    //unfoll anonymous user
+    const targetUser = await UsersFunction.findUserById(User, followedUser);
+    if (targetUser.is_anonymous === false) {
+      const anonymousUser = await UsersFunction.findAnonymousUserId(User, followedUser);
+      await unfollUser(
+        userFollowUserModel,
+        userFollowUserHistoryModel,
+        userId,
+        anonymousUser.user_id,
+        followSource,
+        t
+      );
+    }
+    //end unfoll anonymous user
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential performance issue. You're making two separate calls to `unfollUser` function, one for signed users and another for anonymous users. Each call involves a database transaction, which can be expensive. If possible, consider refactoring your code to perform both operations within a single transaction. This would reduce the number of transactions and potentially improve performance. However, this might require significant changes to your code structure and logic, so please consider it carefully.


<!-- This is an auto-generated comment by OSS CodeRabbit -->